timeseries/src/autogluon/timeseries/predictor.py

```diff
@@ -205,6 +205,11 @@ def _check_and_prepare_data_frame(self, df: TimeSeriesDataFrame) -> TimeSeriesDa
         """
         if df is None:
             return df
+        if not isinstance(df, TimeSeriesDataFrame):
```

Comment:
should we document this behavior, and if not -- should we include it? i.e., type coercion has the risk that a user may provide a perfectly good data frame, something may implicitly get converted wrong, and they have no way to inspect what's wrong.

Reply:
We say in documentation that we always expect `TimeSeriesDataFrame` objects as input (i.e., `data`, `train_data`, `tuning_data`, `known_covariates`) but don't perform any checks. 

This means, for example, if a user provides a `pd.DataFrame` without multiindex (that can be seamlessly converted to a TSDF with `TimeSeriesDataFrame(df)`), they will receive a cryptic exception like 
```
KeyError: 'Requested level (timestamp) does not match index name (None)'
```

With the proposed change, we either do the conversion automatically or tell the user 
- that the data must be in TSDF format
- why it cannot be converted to a TSDF automatically (as in the screenshot below)

![Screenshot 2022-12-30 at 13 37 23](https://user-images.githubusercontent.com/6944857/210070931-498c72dd-4578-49df-a4ba-ab4944aeec1a.png)

Reply:
Do you have some scenario in mind where the type conversion may be problematic?

Reply:
Documented the type coercion behavior, as we discussed.